src/Storefront/Resources/views/storefront/base.html.twig

```diff
@@ -36,7 +36,7 @@
         {% endblock %}
 
         {% block base_esi_header %}
-            {{ render_esi(url('frontend.header', { headerParameters: headerParameters })) }}
+            {{ render_esi(url('frontend.header', { headerParameters: headerParameters }), { ignore_errors: false } ) }}
```

Comment:
please do not merge yet, will add a changelog entry, wanted to align on the approach first